packages/ui/src/components/w3m-box-button/index.ts

```diff
@@ -11,13 +11,14 @@ export class W3mBoxButton extends LitElement {
   // -- state & properties ------------------------------------------- //
   @property() public icon?: TemplateResult<2> = undefined
   @property() public label = ''
+  @property() public loading?: boolean = false
   @property() public onClick: () => void = () => null
 
   // -- render ------------------------------------------------------- //
   protected render() {
     return html`
-      <button @click=${this.onClick}>
-        <div>${this.icon}</div>
+      <button @click=${this.onClick} class="${this.loading ? 'disabled' : ''}">
+        <div>${this.loading ? html`<span class="spinner"></span>` : html`${this.icon}`}</div>
```

Comment:
There is existing `w3m-spinner` component that can be re-used here. Might need to adjust it, to take in different size. Will also mean we can remove extra css made for spinner.